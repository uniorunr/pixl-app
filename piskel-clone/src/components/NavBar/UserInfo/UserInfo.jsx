import React from 'react';
import './UserInfo.scss';
import PropTypes from 'prop-types';
import FireBase from '../../../firebase/firebase';

const logout = async () => {
  await FireBase.logout();
};

const UserInfo = ({ src }) => (
  <div className="user-info">
    <img
      className="user-info__avatar"
      src={src}
      alt="User avatar"
      width={27}
      height={27}
    />
    <button className="user-info__sign-out" type="button" onClick={logout}>
      <i className="fas fa-sign-out-alt" />
    </button>
  </div>
);

UserInfo.propTypes = {
  src: PropTypes.string.isRequired,
};

export default UserInfo;
