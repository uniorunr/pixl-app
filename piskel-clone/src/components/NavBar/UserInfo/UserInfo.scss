.user-info {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;

  .user-info__avatar {
    border: 2px solid black;
    border-radius: 50px;
  }

  .user-info__sign-out {
    margin: 0 0 0 10px;
    padding: 4px 4px 2px;
    font-size: 1.6rem;
    background-color: white;
    border: none;
    cursor: pointer;
    outline: none;
  }

  .user-info__sign-out:active {
    border: 2px solid black;
  }
}
