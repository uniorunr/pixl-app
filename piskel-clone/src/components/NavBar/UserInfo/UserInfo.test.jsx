import React from 'react';
import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';

it('renders correctly', () => {
  const tree = renderer.create(<UserInfo src="src" />).toJSON();
  expect(tree).toMatchSnapshot();
});
