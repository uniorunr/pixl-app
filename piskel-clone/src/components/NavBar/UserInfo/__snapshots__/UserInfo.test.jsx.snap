// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user-info"
>
  <img
    alt="User avatar"
    className="user-info__avatar"
    height={27}
    src="src"
    width={27}
  />
  <button
    className="user-info__sign-out"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-sign-out-alt"
    />
  </button>
</div>
`;
